### PR TITLE
#96-cge-pandas-warning

### LINKED ISSUE #96: Pandas warning - default dtype for empty series will be 'object' instead of 'float64' in a future version
joplin_cge.ipynb and seaside_cge.ipynb under incore-docs/notebooks has warning from pandas

> FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.

the dump of warning message is in attached.

[pandas-warning.log](https://github.com/IN-CORE/pyincore/files/8044002/pandas-warning.log)

### PR DESCRIPTION
**Pull request by diegoac2**
added explicit data type to pandas series since next versions will default to object and that would break the analysis 
closes #96

### CODE REVIEW

**Comment by navarroc** [comment]
@diegoac2 I also get dtype warnings from equationlib.py, can you fix those as well?

**Comment by navarroc** [comment]
Similar with seasidecge, it has an equationlib.py file that has some dtype warnings. Also the pandas.append warnings.

File: pyincore/analyses/seasidecge/seasidecge.py [hunk 1]
```diff
@@ -875,7 +875,7 @@ def _(x):
 
         # create data frame for factor taxes by sector
 
-        a = pd.Series(index=I).fillna(0.0)
+        a = pd.Series(index=I, dtype='float64').fillna(0.0)
         a = SAM.loc[USSOCL, I].append(a, ignore_index=True).append(SAM.loc[GL, I])  # labor, land, capital
```
Comment (navarroc) [inline pyincore/analyses/seasidecge/seasidecge.py:879]: this append needs to be replaced by concat

File: pyincore/analyses/joplincge/joplincge.py [hunk 2]
```diff
@@ -537,10 +537,9 @@ def run(self):
 
         out.loc[G, G] = IOUT.loc[G, G]
 
-        BETA = pd.concat([MISC.loc[I, ['ETAY']]] * len(H), axis=1)
+        BETA = pd.append([MISC.loc[I, ['ETAY']]] * len(H), axis=1)
```
Comment (navarroc) [inline pyincore/analyses/joplincge/joplincge.py:540]: this should be concat

File: pyincore/analyses/joplincge/joplincge.py [hunk 3]
```diff
@@ -584,7 +583,7 @@ def run(self):
         ETAL = LANDCAP.loc[IG, ['ETAL1']].T
         ETAL.index = LA
 
-        ETALB = pd.concat([ETALB1] * len(L), axis=1)
+        ETALB = pd.append([ETALB1] * len(L), axis=1)
```
Comment (navarroc) [inline pyincore/analyses/joplincge/joplincge.py:586]: this should also be concat

File: pyincore/analyses/joplincge/joplincge.py [hunk 4]
```diff
@@ -785,7 +784,7 @@ def run(self):
         # RA0.loc[F] = 1.0
 
         # create data frame for factor taxes by sector
-        a = pd.Series(index=I).fillna(0.0)
+        a = pd.Series(index=I, dtype='float64').fillna(0.0)
         a = SAM.loc[USSOCL, I].append(a, ignore_index=True).append(SAM.loc[GL, I])  # labor, land, capital
```
Comment (navarroc) [inline pyincore/analyses/joplincge/joplincge.py:788]: these appends need to be concat

**Comment by navarroc** [comment]
@diegoac2 Are you able to test these changes on your local? I manually copy the updated python files into a conda environment so I can run with the notebooks and see the results. I get the errors/issues noted in my other comments when I try to run these changes locally.

File: pyincore/analyses/joplincge/joplincge.py (hunk 2 above)
Comment (diegoac2) [reply pyincore/analyses/joplincge/joplincge.py:540]: fixed

File: pyincore/analyses/seasidecge/seasidecge.py (hunk 1 above)
Comment (diegoac2) [reply pyincore/analyses/seasidecge/seasidecge.py]: fixed

File: pyincore/analyses/joplincge/joplincge.py (hunk 4 above)
Comment (diegoac2) [reply pyincore/analyses/joplincge/joplincge.py]: fixed

File: pyincore/analyses/joplincge/joplincge.py (hunk 3 above)
Comment (diegoac2) [reply pyincore/analyses/joplincge/joplincge.py]: fixed

**Comment by navarroc** [comment]
When I run joplin_cge.ipynb, I get the following error:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [7], in <module>
      1 # Run Joplin CGE model analysis
----> 2 joplin_cge.run_analysis()

File ~/miniconda3/envs/pyincore-1.3.0/lib/python3.8/site-packages/pyincore/baseanalysis.py:264, in BaseAnalysis.run_analysis(self)
    261         print("Error reading parameter: " + result[1])
    262         return result
--> 264 return self.run()

File ~/miniconda3/envs/pyincore-1.3.0/lib/python3.8/site-packages/pyincore/analyses/joplincge/joplincge.py:788, in JoplinCGEModel.run(self)
    784 # RA0.loc[F] = 1.0
    785 
    786 # create data frame for factor taxes by sector
    787 a = pd.Series(index=I, dtype='float64').fillna(0.0)
--> 788 a = SAM.loc[USSOCL, I].concat(a, ignore_index=True).append(SAM.loc[GL, I])  # labor, land, capital
    789 a.index = F
    791 ALPHA.loc[F, I] = (SAM.loc[F, I] + a.loc[F, I]) / (SAM.loc[F, I].sum(0) + SAM.loc[GF, I].sum(0))

File ~/miniconda3/envs/pyincore-1.3.0/lib/python3.8/site-packages/pandas/core/generic.py:5583, in NDFrame.__getattr__(self, name)
   5576 if (
   5577     name not in self._internal_names_set
   5578     and name not in self._metadata
   5579     and name not in self._accessors
   5580     and self._info_axis._can_hold_identifiers_and_holds_name(name)
   5581 ):
   5582     return self[name]
-> 5583 return object.__getattribute__(self, name)

AttributeError: 'DataFrame' object has no attribute 'concat'

**Comment by navarroc** [comment]
I think the issue is you can't just replace append with concat, they are slightly different functions. You have to do something like:

array1 = pd.concat[array1, array2]

**Review by ncsa-mo — APPROVED** [review]

**Review by longshuicy — APPROVED** [review]

**Pull request merged**